src/main/java/org/spongepowered/common/event/tracking/phase/packet/PacketFunction.java

```diff
@@ -957,13 +957,33 @@ static void processEntities(EntityPlayerMP player, CauseTracker causeTracker, Co
             default:
                 throw new AssertionError();
         }
+        SpongeImpl.postEvent(SpongeEventFactory.createResourcePackStatusEvent(Cause.source(player).build(),pack, (Player) player, status));
         if (status.wasSuccessful().isPresent()) {
-            mixinHandler.getSentResourcePacks().remove(hash);
+            mixinHandler.getPendingResourcePackQueue().remove();
+
+            if (!mixinHandler.getPendingResourcePackQueue().isEmpty()) {
+                Cause supersededCause = Cause.source(player).named(InternalNamedCauses.Packet.RESPONDED_RESOURCE_PACK, pack).build();
+                while (mixinHandler.getPendingResourcePackQueue().size() > 1) {
```

Comment:
Yes.

- If there is only 1 resource pack in the queue at the start of the method, it will be removed by this point.
- If there are 2 resource packs in the queue, there will now be only one, and that one needs to be sent to the client (done after the `while` loop), so the loop does not run.
- If there are 3 or more resource packs in the queue, any plugins that are listening to those resource packs need to be notified, so the `while` loop runs to clean those up.